tests/Unit/CrudPanel/CrudPanelColumnsTest.php

```diff
@@ -0,0 +1,242 @@
+
+        $this->assertEquals(array_keys($this->expectedTwoColumnsArray), array_keys($this->crudPanel->columns));
+    }
+
+    public function testRemoveColumnByName()
+    {
+        $this->markTestIncomplete('Not correctly implemented');
+
+        // TODO: fix the remove column functionality
+        $this->crudPanel->addColumns(['column1', 'column2', 'column3']);
+        $this->crudPanel->removeColumn('column1');
+
+        $this->assertEquals(2, count($this->crudPanel->columns));
+        $this->assertNotContains('column1', $this->crudPanel->columns);
+        $this->assertNotContains($this->oneColumnArray, $this->crudPanel->columns);
+    }
+
+    public function testRemoveUnknownColumnName()
+    {
+        $this->markTestIncomplete('Not correctly implemented');
+
+        // TODO: fix the remove column functionality
+        $this->crudPanel->addColumns(['column1', 'column2', 'column3']);
+
+        // TODO: should this fail with an exception or just log as warning?
+        $this->crudPanel->removeColumn('column4');
+
+        $this->assertEquals(3, count($this->crudPanel->columns));
+        $this->assertNotContains('column4', $this->crudPanel->columns);
+        $this->assertNotContains($this->otherOneColumnArray, $this->crudPanel->columns);
+    }
+
+    public function testRemoveColumns()
+    {
+        $this->markTestIncomplete('Not correctly implemented');
+
+        // TODO: fix the remove column functionality
+        $this->crudPanel->addColumns(['column1', 'column2', 'column3']);
+        $this->crudPanel->removeColumns($this->twoColumnArray);
+
+        $this->assertEquals(2, count($this->crudPanel->columns));
+        $this->assertNotContains(['column1', 'column2'], $this->crudPanel->columns);
+        $this->assertNotEquals($this->expectedThreeColumnsArray, $this->crudPanel->columns);
+    }
+
+    public function testRemoveUnknownColumns()
+    {
+        $this->markTestIncomplete('Not correctly implemented');
+
+        // TODO: fix the remove column functionality
+        $this->crudPanel->addColumns(['column1', 'column2', 'column3']);
+
+        // TODO: should this fail with an exception or just log as warning?
+        $this->crudPanel->removeColumn($this->otherOneColumnArray);
+
+        $this->assertEquals(3, count($this->crudPanel->columns));
+        $this->assertNotContains('column4', $this->crudPanel->columns);
+        $this->assertNotContains($this->otherOneColumnArray, $this->crudPanel->columns);
+    }
+
+    public function testSetColumnDetails()
+    {
+        $this->markTestIncomplete('Not correctly implemented');
+
+        // TODO: refactor crud panel sync method
+    }
+
+    public function testSetColumnsDetails()
+    {
+        $this->markTestIncomplete('Not correctly implemented');
+
+        // TODO: refactor crud panel sync method
+    }
+
+    public function testSetColumnOrder()
+    {
+        $this->markTestIncomplete('Not yet implemented');
```

Comment:
What's the difference between correctly and not yet here?

(also, the test names are horridly close to each other around here - I had to squit to realise one's plural and the other's not - call me 👴).

Reply:
Those were methods that didn't to what they were expected to do. The `setColumnOrder` is literally an empty method.